pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -485,19 +556,52 @@ class HassTypeHintChecker(BaseChecker):  # type: ignore[misc]
     def __init__(self, linter: PyLinter | None = None) -> None:
         super().__init__(linter)
         self._function_matchers: list[TypeHintMatch] = []
+        self._class_matchers: list[ClassTypeHintMatch] = []
 
     def visit_module(self, node: astroid.Module) -> None:
         """Called when a Module node is visited."""
         self._function_matchers = []
+        self._class_matchers = []
 
         if (module_platform := _get_module_platform(node.name)) is None:
             return
 
         if module_platform in _PLATFORMS:
             self._function_matchers.extend(_FUNCTION_MATCH["__any_platform__"])
 
-        if matches := _FUNCTION_MATCH.get(module_platform):
-            self._function_matchers.extend(matches)
+        if function_matches := _FUNCTION_MATCH.get(module_platform):
+            self._function_matchers.extend(function_matches)
+
+        if class_matches := _CLASS_MATCH.get(module_platform):
+            self._class_matchers = class_matches
+
+    def visit_classdef(self, node: astroid.ClassDef) -> None:
+        """Called when a ClassDef node is visited."""
+        for class_matches in self._class_matchers:
+            for base in node.bases:
+                if (
+                    isinstance(base, astroid.Name)
+                    and base.name in class_matches.classes
+                ) or (
+                    isinstance(base, astroid.Attribute)
+                    and base.attrname in class_matches.classes
+                ):
+                    self._visit_class_functions(node, class_matches.matches)
+
+    def _visit_class_functions(
+        self, node: astroid.ClassDef, matches: list[TypeHintMatch]
+    ) -> None:
+        for match in matches:
+            for function_node in node.mymethods():
+                function_name: str | None = function_node.name
+                if match.function_name == function_name:
+                    self._check_function(function_node, match)
+                elif (
+                    function_name is not None
+                    and match.function_name.endswith("*")
+                    and function_name.startswith(match.function_name[:-1])
+                ):
```

Comment:
The `elif` case isn't used (yet), is it? Maybe it's better to defer adding it until it's needed.